src/Ducks/MamaDuck.cpp

```diff
@@ -88,6 +88,56 @@ void MamaDuck::handleReceivedPacket() {
       return;
     }
 
+    if (rxPacket->getTopic() == topics::receipt) {
+      CdpPacket packet = CdpPacket(rxPacket->getBuffer());
+
+      if (duid[0] == packet.dduid[0]
+        && duid[1] == packet.dduid[1]
+        && duid[2] == packet.dduid[2]
+        && duid[3] == packet.dduid[3]
+        && duid[4] == packet.dduid[4]
+        && duid[5] == packet.dduid[5]
+        && duid[6] == packet.dduid[6]
+        && duid[7] == packet.dduid[7]
+      ) {
+        if (lastMessageMuid.size() != MUID_LENGTH) {

```

Comment:
MUID is good enough for small networks now, but if we plan on supporting thousands of people in a disaster scenario, we should use a proper hash function to avoid collisions. Hashes can be long, but they provide good guarantees.

Reply:
Currently my goal is to implement ACK broadcasts from the Papa, which will broadcast DUID/MUID pairs. So there shouldn't be any MUID collisions with that scheme. (Technically a duck may have MUIDs collide within itself, but that would be so rare that other problems would matter orders of magnitude more, like packet loss due to congestion or due to interference.)

Trello card is here: https://trello.com/c/32yHLXEn/20-broadcast-ack-reservedtopic